python/pydantic_core/core_schema.py

```diff
@@ -1384,6 +1384,24 @@ def uuid_schema(
     )
 
 
+class NestedModelSchema(TypedDict, total=False):
+    type: Required[Literal['nested-model']]
+    model: Required[Type[Any]]
+    metadata: Any
+
+
+def nested_model_schema(
+    *,
+    model: Type[Any],
+    metadata: Any = None,
```

Comment:
```suggestion
    metadata: Dict[str, Any] | None = None,
```

Reply:
According to a recent change I made - probably best to rebase with `main` just for ease of review